routes/publish.js

```diff
@@ -0,0 +1,71 @@
+var request = require("request");
+
+function isEmpty(val) {
+  return val === null || val === undefined;
+}
+
+module.exports = function(config) {
+  return function(req, res) {
+    var project = JSON.parse(JSON.stringify(req.session.project.meta));
+    delete project.id;
+    delete project.publish_url;
+
+    if(!req.body || isEmpty(req.body.description) || isEmpty(req.body.dateUpdated) || isEmpty(req.body.public)) {
+      res.status(400).send({ error:  "Request body missing data" });
+      return;
+    }
+
+    project.description = req.body.description;
+    // Uncomment the line below once https://github.com/mozilla/publish.webmaker.org/issues/98 is done
+    // project.public = req.body.public;
+    project.date_updated = req.body.dateUpdated;
+
+    request({
+      method: "PUT",
+      uri: config.publishURL + "/projects/" + req.session.project.meta.id,
```

Comment:
So every `publish` call from thimble includes the entire project? Do we really need to do this? I feel like the front end should prevent a publish action if the state of the project is dirty.